scripts/screens/ChangeGenderScreen.py

```diff
@@ -0,0 +1,631 @@
+        self.buttons = {}
+        self.next_cat = None
+        self.previous_cat = None
+        self.elements = {}
+        self.windows = None
+        self.removalboxes_text = {}
+        self.removalbuttons = {}
+        self.pronoun_template = [
+            {
+                "ID": "Custom",
+                "subject": "",
+                "object": "",
+                "poss": "",
+                "inposs": "",
+                "self": "",
+                "conju": 1
+            }
+        ]
+        self.remove_button = {}
+        self.removalboxes_text = {}
+        self.boxes = {}
+        self.box_labels = {}
+        self.conju = 2
+
+    def handle_event(self, event):
+        if event.type == pygame_gui.UI_BUTTON_START_PRESS:
+            if event.ui_element == self.back_button:
+                self.change_screen("profile screen")
+            elif event.ui_element == self.next_cat_button:
+                if isinstance(Cat.fetch_cat(self.next_cat), Cat):
+                    game.switches["cat"] = self.next_cat
+                    self.update_selected_cat()
+                # else:
+                # print("invalid next cat", self.next_cat)
+            elif event.ui_element == self.previous_cat_button:
+                if isinstance(Cat.fetch_cat(self.previous_cat), Cat):
+                    game.switches["cat"] = self.previous_cat
+                    self.update_selected_cat()
+                # else:
+                # print("invalid previous cat", self.previous_cat)
+            elif event.ui_element == self.buttons["save"]:
+                if self.are_boxes_full():
+                    gender_identity = self.get_new_identity()
+                    self.the_cat.genderalign = gender_identity
+                    print("New Gender Identity Unlocked!")
+                    self.selected_cat_elements["identity_changed"].show()
+                    self.selected_cat_elements["cat_gender"].kill()
+                    self.selected_cat_elements["cat_gender"] = pygame_gui.elements.UITextBox(
+                        f"{self.the_cat.genderalign}",
+                        scale(pygame.Rect((252, 500), (500, 500))),
+                        object_id=get_text_box_theme("#text_box_30_horizcenter_spacing_95"),
+                        manager=MANAGER
+                    )
+
+            elif event.ui_element == self.buttons["add_pronouns"]:
+                PronounCreation(self.the_cat)
+                self.previous_cat_button.disable()
+                self.next_cat_button.disable()
+                self.back_button.disable()
+
+            else:
+                with open('resources/dicts/pronouns.json', 'r', encoding='utf-8') as file:
+                    pronouns_data = ujson.load(file)
+
+                for pronounset in pronouns_data["default_pronouns"]:
+                    add_button_id = f"add_button_{pronounset['ID']}" + "preset"
+                    if event.ui_element == self.buttons.get(add_button_id):
```

Comment:
Rather than looping button keys, I would suggest using something similar to CatButton or UISpriteButton, which adds an additional data field to the button object, which can then be extracted from the button event. In fact, you would probably add an object_id field to the CatButton constructor and use that class directly.  This would also help remove the need for the 'ID' value for pronoun dictionaries.

Reply:
I believe this is part of what you're helping with, so I leave it in your capable hands  :D